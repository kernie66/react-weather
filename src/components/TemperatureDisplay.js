import Temperature from './Temperature';
import MinMax from './MinMax';
import { Col, Container } from 'reactstrap';
import ErrorBoundary from './ErrorBoundary';

export default function TemperatureDisplay() {
  return (
    <Container className="d-flex justify-content-evenly m-0" style={{ color: "sandybrown" }}>
      <Col xs="auto">
        <ErrorBoundary>
          <Temperature />
        </ErrorBoundary>
      </Col>
      <Col xs="auto">
        <ErrorBoundary>
          <MinMax />
        </ErrorBoundary>
      </Col>
    </Container>
  );
};