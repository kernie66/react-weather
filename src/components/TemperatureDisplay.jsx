import Temperature from './Temperature';
import MinMax from './MinMax';
import ErrorBoundary from './ErrorBoundary';
import { Group } from '@mantine/core';

export default function TemperatureDisplay() {
  return (
    <Group
      justify="space-between"
      mx="lg"
      style={{ color: 'sandybrown' }}
    >
      <ErrorBoundary>
        <Temperature />
      </ErrorBoundary>
      <ErrorBoundary>
        <MinMax />
      </ErrorBoundary>
    </Group>
  );
}
