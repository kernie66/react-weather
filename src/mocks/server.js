import { setupServer } from 'msw/node';
import { weatherHandlers } from './weatherHandlers.js';
import { defaultHandlers } from './defaultHandlers.js';
import { translationHandlers } from './translationHandlers.js';

export const server = setupServer(
  ...weatherHandlers,
  ...translationHandlers,
  ...defaultHandlers
);
