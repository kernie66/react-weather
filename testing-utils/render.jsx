// ./test-utils/render.tsx
import { render as testingLibraryRender } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
// Import your theme object
import { createTheme } from '@mantine/core';
import {
  QueryClient,
  QueryClientProvider,
} from '@tanstack/react-query';

const theme = createTheme({});

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
    logger: {
      log: console.log,
      warn: console.warn,
      error: () => {},
    },
  });

export const testQueryClient = createTestQueryClient();

export function render(ui) {
  console.log('testQueryClient', testQueryClient);
  return testingLibraryRender(<>{ui}</>, {
    wrapper: ({ children }) => (
      <MantineProvider theme={theme}>
        <QueryClientProvider client={testQueryClient}>
          {children}
        </QueryClientProvider>
      </MantineProvider>
    ),
  });
}
