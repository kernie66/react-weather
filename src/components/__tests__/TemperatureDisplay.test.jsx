import { http, HttpResponse } from 'msw';
import {
  render,
  renderWithNotifications,
  screen,
  userEvent,
  waitForElementToBeRemoved,
} from '../../../testing-utils';
import { server } from '../../mocks/server.js';
import TemperatureDisplay from '../TemperatureDisplay.jsx';
import { expect } from 'vitest';
import { omit } from 'radash';
import weatherDataJson from '../../../bruno/weather_alert_response.json';
import { testQueryClient } from '../../../testing-utils/render.jsx';

const baseURL = import.meta.env.VITE_BASE_URL;

describe('test TemperatureDisplay', () => {
  it('renders the main temperature display area with weather alert', async () => {
    const user = userEvent.setup();

    const snapshot = renderWithNotifications(<TemperatureDisplay />);

    // Weather alert button
    const alertButton = await screen.findByRole('button', {
      name: /vädervarning/i,
    });
    expect(alertButton).toBeInTheDocument();

    // Current temperature
    expect(await screen.findByText('19')).toBeInTheDocument();
    expect(await screen.findByText(/\.2°C/)).toBeInTheDocument();
    // Max temperature
    expect(await screen.findByText('20')).toBeInTheDocument();
    expect(await screen.findByText(/\.9°C/)).toBeInTheDocument();
    // Min temperature
    expect(await screen.findByText('12')).toBeInTheDocument();
    expect(await screen.findByText(/\.0°C/)).toBeInTheDocument();
    // Notification
    expect(await screen.findByRole('alert')).toBeInTheDocument();
    expect(
      await screen.queryByText(/kategori/i)
    ).not.toBeInTheDocument();
    expect(await screen.findByText(/regnskur/i)).toBeInTheDocument();
    expect(
      await screen.findByText(/translated text: 127 words/i)
    ).toBeInTheDocument();
    expect(snapshot).toMatchSnapshot();

    // Open weather alert modal, should remove notification
    await user.click(alertButton);
    await waitForElementToBeRemoved(() =>
      screen.queryByText(/vädernotifiering/i)
    );
    console.log('Notification removed');
    expect(await screen.findByText(/kategori/i)).toBeInTheDocument();
    expect(await screen.findByText(/källa/i)).toBeInTheDocument();
    expect(await screen.findByText(/regnskur/i)).toBeInTheDocument();
    expect(
      await screen.findByText(/translated text: 127 words/i)
    ).toBeInTheDocument();
  });

  it('renders the main temperature display area without weather alert', async () => {
    const weatherDataNoAlert = omit(weatherDataJson, ['alerts']);
    server.use(
      http.get(`${baseURL}/onecall`, () => {
        console.log('Getting weather data without alert');
        return HttpResponse.json(weatherDataNoAlert);
      })
    );
    // Ensure that a new query is fetched
    testQueryClient.invalidateQueries();
    render(<TemperatureDisplay />);

    // Current temperature
    expect(await screen.findByText('19')).toBeInTheDocument();
    expect(await screen.findByText(/\.2°C/)).toBeInTheDocument();
    // Max temperature
    expect(await screen.findByText('20')).toBeInTheDocument();
    expect(await screen.findByText(/\.9°C/)).toBeInTheDocument();
    // Min temperature
    expect(await screen.findByText('12')).toBeInTheDocument();
    expect(await screen.findByText(/\.0°C/)).toBeInTheDocument();
    // No weather alert button
    expect(
      screen.queryByRole('button', { name: /vädervarning/i })
    ).not.toBeInTheDocument();
  });
});
