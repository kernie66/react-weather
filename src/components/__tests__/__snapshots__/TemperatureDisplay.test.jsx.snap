// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test TemperatureDisplay > renders the main temperature display area with weather alert 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: calc(2.25rem * var(--mantine-scale)); --group-align: center; --group-justify: space-between; --group-wrap: wrap; color: sandybrown;"
      >
        <div
          class="weather-alert--box"
          style="min-width: calc(4rem * var(--mantine-scale));"
        >
          <button
            class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
            data-size="xl"
            data-variant="transparent"
            style="--button-height: var(--button-height-xl); --button-padding-x: var(--button-padding-x-xl); --button-fz: var(--mantine-font-size-xl); --button-bg: transparent; --button-hover: transparent; --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem;"
            type="button"
          >
            <span
              class="m_80f1301b mantine-Button-inner"
            >
              <span
                class="m_811560b9 mantine-Button-label"
              >
                <svg
                  color="orange"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  style="color: orange;"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 9v4"
                  />
                  <path
                    d="M10.363 3.591l-8.106 13.534a1.914 1.914 0 0 0 1.636 2.871h16.214a1.914 1.914 0 0 0 1.636 -2.87l-8.106 -13.536a1.914 1.914 0 0 0 -3.274 0z"
                  />
                  <path
                    d="M12 16h.01"
                  />
                </svg>
                <span
                  class="m_515a97f8 mantine-VisuallyHidden-root"
                >
                  Vädervarning
                </span>
              </span>
            </span>
          </button>
        </div>
        <p
          class="mantine-focus-auto _outlineDouble_597f95 m_b6d8b162 mantine-Text-root"
          style="margin-inline-start: auto; font-weight: 500; text-align: right;"
        >
          19
          <span
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-size: 80%; font-weight: 500;"
          >
            .2°C
          </span>
        </p>
        <div
          class="m_6d731127 mantine-Stack-root"
          style="--stack-gap: 0rem; --stack-align: stretch; --stack-justify: center;"
        >
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            style="margin-inline-start: auto; font-weight: 500; text-align: right;"
          >
            20
            <span
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="font-size: 80%; font-weight: 500;"
            >
              .9°C
            </span>
          </p>
          <div
            class="m_3eebeb36 mantine-Divider-root"
            data-orientation="horizontal"
            data-size="md"
            role="separator"
            style="--divider-size: var(--divider-size-md);"
          />
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            style="margin-inline-start: auto; font-weight: 500; text-align: right;"
          >
            12
            <span
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="font-size: 80%; font-weight: 500;"
            >
              .0°C
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      data-portal="true"
    >
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="top-center"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="top-left"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="top-right"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="bottom-right"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div>
          <div
            class="m_5ed0edd0 mantine-Notifications-notification m_a513464 mantine-Notification-root"
            data-with-border="true"
            data-with-icon="true"
            id="notification-alert"
            role="alert"
            style="--notification-color: var(--mantine-color-orange-filled); opacity: 1; max-height: 200px; transform: translateX(0); transition-duration: 250ms, 250ms, 250ms; transition-timing-function: cubic-bezier(.51,.3,0,1.21), cubic-bezier(.51,.3,0,1.21), linear; transition-property: opacity, transform, max-height;"
          >
            <div
              class="m_a4ceffb mantine-Notification-icon"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 9v4"
                />
                <path
                  d="M10.363 3.591l-8.106 13.534a1.914 1.914 0 0 0 1.636 2.871h16.214a1.914 1.914 0 0 0 1.636 -2.87l-8.106 -13.536a1.914 1.914 0 0 0 -3.274 0z"
                />
                <path
                  d="M12 16h.01"
                />
              </svg>
            </div>
            <div
              class="m_a49ed24 mantine-Notification-body"
            >
              <div
                class="m_3feedf16 mantine-Notification-title"
              >
                Vädervarning: Regnskur
              </div>
              <div
                class="m_3d733a3a mantine-Notification-description"
                data-with-title="true"
              >
                <div
                  class="m_6d731127 mantine-Stack-root"
                  style="--stack-gap: calc(0.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
                >
                  <p
                    class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                    data-size="sm"
                    style="--text-fz: var(--mantine-font-size-sm); --text-lh: var(--mantine-line-height-sm); color: dodgerblue;"
                  >
                    Från: 
                    lördag 13 juli 2024 kl. 02:00
                  </p>
                  <p
                    class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                    data-size="sm"
                    style="--text-fz: var(--mantine-font-size-sm); --text-lh: var(--mantine-line-height-sm); color: dodgerblue;"
                  >
                    Till: 
                    lördag 13 juli 2024 kl. 21:00
                  </p>
                  <p
                    class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                    data-line-clamp="true"
                    data-size="sm"
                    style="--text-fz: var(--mantine-font-size-sm); --text-lh: var(--mantine-line-height-sm); --text-line-clamp: 3; white-space: pre-wrap;"
                  >
                    Translated text: 127 words
                  </p>
                  <span
                    class="m_515a97f8 mantine-VisuallyHidden-root"
                  >
                    Vädernotifiering
                  </span>
                </div>
              </div>
            </div>
            <button
              class="mantine-focus-auto mantine-active m_220c80f2 m_919a4d88 mantine-Notification-closeButton m_86a44da5 mantine-CloseButton-root m_87cf2631 mantine-UnstyledButton-root"
              color="gray"
              data-variant="subtle"
              style="--cb-icon-size: calc(1rem * var(--mantine-scale));"
              type="button"
            >
              <svg
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="bottom-left"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
      <div
        class="m_b37d9ac7 mantine-Notifications-root"
        data-position="bottom-center"
        style="--notifications-z-index: 400; --notifications-container-width: calc(27.5rem * var(--mantine-scale));"
      >
        <div />
      </div>
    </div>
    <div
      data-portal="true"
    >
      <div>
        <div
          class="m_9df02822 mantine-Modal-root"
          data-size="xl"
          style="--modal-size: var(--modal-size-xl); --modal-y-offset: 5dvh; --mb-z-index: 200;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div
      class="m_4081bf90 mantine-Group-root"
      style="--group-gap: calc(2.25rem * var(--mantine-scale)); --group-align: center; --group-justify: space-between; --group-wrap: wrap; color: sandybrown;"
    >
      <div
        class="weather-alert--box"
        style="min-width: calc(4rem * var(--mantine-scale));"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="xl"
          data-variant="transparent"
          style="--button-height: var(--button-height-xl); --button-padding-x: var(--button-padding-x-xl); --button-fz: var(--mantine-font-size-xl); --button-bg: transparent; --button-hover: transparent; --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                color="orange"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                style="color: orange;"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 9v4"
                />
                <path
                  d="M10.363 3.591l-8.106 13.534a1.914 1.914 0 0 0 1.636 2.871h16.214a1.914 1.914 0 0 0 1.636 -2.87l-8.106 -13.536a1.914 1.914 0 0 0 -3.274 0z"
                />
                <path
                  d="M12 16h.01"
                />
              </svg>
              <span
                class="m_515a97f8 mantine-VisuallyHidden-root"
              >
                Vädervarning
              </span>
            </span>
          </span>
        </button>
      </div>
      <p
        class="mantine-focus-auto _outlineDouble_597f95 m_b6d8b162 mantine-Text-root"
        style="margin-inline-start: auto; font-weight: 500; text-align: right;"
      >
        19
        <span
          class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          style="font-size: 80%; font-weight: 500;"
        >
          .2°C
        </span>
      </p>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: 0rem; --stack-align: stretch; --stack-justify: center;"
      >
        <p
          class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          style="margin-inline-start: auto; font-weight: 500; text-align: right;"
        >
          20
          <span
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-size: 80%; font-weight: 500;"
          >
            .9°C
          </span>
        </p>
        <div
          class="m_3eebeb36 mantine-Divider-root"
          data-orientation="horizontal"
          data-size="md"
          role="separator"
          style="--divider-size: var(--divider-size-md);"
        />
        <p
          class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          style="margin-inline-start: auto; font-weight: 500; text-align: right;"
        >
          12
          <span
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-size: 80%; font-weight: 500;"
          >
            .0°C
          </span>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
